async_cache.go

```diff
@@ -29,18 +30,55 @@ type keyStatus struct {
 
 //Async Cache
 type AsyncCache struct {
-	gCache    *Cache
+	*Cache
 	Fetcher   *Fetcher
 	keystatus *keyStatus
+	errorFunc func(key string, err error)
+}
+
+type Config struct {
+	Fetcher             *Fetcher
+	PurgeTime           time.Duration
+	ExpiryTime          time.Duration
+	ErrorFuncDefination func(key string, err error)
```

Comment:
Change to ErrorfuncDefination to ErrorFunc in Config

Remove Reference  for *Config
Return for nil for aConfig.nil